Update DeveloperGuide and UserGuide with borrow/return commands (#92)
AY2223S2-CS2103T-T12-3/tp: docs/UserGuide.md

Format: `listBook`

### Borrowing a book : `borrow`

> Lend a book to a person given a book ID, person ID and due date.

Reviewer: ```suggestion
Lends a book to a person given a person index, book index and due date.
```

> Format: `borrow PERSON_ID b/BOOK_INDEX d/DUE_DATE`

Reviewer: ```suggestion
Format: `borrow PERSON_INDEX b/BOOK_INDEX d/DUE_DATE`
```

> * Lends a book specified by the BOOK_INDEX to the person at the specified PERSON_ID.

Reviewer: ```suggestion
* Lends a book specified by the BOOK_INDEX to the person at the specified PERSON_INDEX.
```

> * The PERSON_ID index refers to the index number shown in the displayed person list.

Reviewer: ```suggestion
* The PERSON_INDEX refers to the index number shown in the displayed person list.
```

* The BOOK_INDEX refers to the index number shown in the displayed book list.

> * The index must be a positive integer 1, 2, 3, …​

Reviewer: ```suggestion
* The indexes must be positive integers 1, 2, 3, …​
```

Examples:

> * `list` followed by `borrow 2 b/3 d/17/10/2024` lends the 3rd book in the book list to the 2nd person in the user records with a due date of Oct. 17, 2024.

Reviewer: ```suggestion
* `listBook/listPerson` followed by `borrow 2 b/3 d/17/10/2024` lends the 3rd book in the book list to the 2nd person in the user records with a due date of Oct. 17, 2024.
```

### Returning a book : `return`

> Return a person's borrowed book to the library given a book ID and person ID.

Reviewer: ```suggestion
Return a person's borrowed book to the library given a person index and book index.
```

> Format: `return PERSON_ID b/BOOK_INDEX`

Reviewer: ```suggestion
Format: `return PERSON_INDEX b/BOOK_INDEX`
```

> * Returns a person's borrowed book specified by the BOOK_INDEX and specified PERSON_ID.

Reviewer: ```suggestion
* Returns a person's borrowed book specified by the PERSON_INDEX and specified BOOK_INDEX.
```

> * The PERSON_ID index refers to the index number shown in the displayed person list.

Reviewer: ```suggestion
* The PERSON_INDEX refers to the index number shown in the displayed person list.
```

* The BOOK_INDEX refers to the index number shown in the displayed book list.

> * The index must be a positive integer 1, 2, 3, …​

Reviewer: ```suggestion
* The indexes must be positive integers 1, 2, 3, …​
```

Examples:
* `list` followed by `return 1 b/2` returns the 2nd book in the book list from the 1st person in the user records.

### Common Functions

> ### Clearing all entries : `clear`

Reviewer: Can you create a new General table in the command summary at the bottom of the UG and add the return and borrow commands there? Thanks.

--------|------------------
**AddPerson** | `addPerson n/NAME p/PHONE_NUMBER e/EMAIL a/ADDRESS [t/TAG]…​` <br> e.g., `add n/James Ho p/22224444 e/jamesho@example.com a/123, Clementi Rd, 1234665 t/friend t/colleague`
**Clear** | `clear`
**Delete** | `delete INDEX`<br> e.g., `delete 3`
**Edit** | `edit INDEX [n/NAME] [p/PHONE_NUMBER] [e/EMAIL] [a/ADDRESS] [t/TAG]…​`<br> e.g.,`edit 2 n/James Lee e/jameslee@example.com`
**Find** | `find KEYWORD [MORE_KEYWORDS]`<br> e.g., `find James Jake`
**List** | `list`

> **Borrow** | `borrow PERSON_INDEX b/BOOK_INDEX d/DUE_DATE` <br> e.g., `borrow 1 b/2 d/23/09/2024` 

Reviewer: Dont need to add the other commands just add Borrow and Return in a separate table below the Book table
